### PR TITLE
♻️  🏗  Create proxy for kleur/colors to expand typing.

### PR DESCRIPTION
**Pull request by rileyajones**
build-system uses the library `kleur` widely for coloring command line outputs, however, the default typing causes some issues for our type-checker. By importing a proxy for the library we can provide global type overrides while still retaining type safety.

### CODE REVIEW

File: build-system/common/colors.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    if (input instanceof Array) {
+      return /** @type {function} */ (formatter)(input.join(' '));
+    }
+
+    return /** @type {function} */ (formatter)(input);
+  };
+  return map;
+}, /**@type {Record<keyof typeof colors, (input?: string | boolean | number | string[] | number[] | boolean []) => string>}   */ ({}));
```
Comment (rsimha) [inline build-system/common/colors.js:36]: While this function addresses all type errors discovered by `amp check-build-system`, I suspect it is doing too much. Things like error checking, handling `null` / `undefined`, formatting of arrays / objects, etc. should be left to the underlying library. Doing it here could prove brittle when future updates to the library introduce subtle changes.

I played around with this code on my local branch and came up with a more concise alternative:

```suggestion
/**
 * @fileoverview The kleur/color library provides a variety of untyped color
 * formatting functions. This file provides a typed wrapper for each of them in
 * order to satisfy our type checks without changing any functionality. For more
 * info, see https://github.com/lukeed/kleur/blob/master/colors.{mjs,d.ts}.
 */

module.exports = Object.entries(colors).reduce((map, [key, formatter]) => {
  map[key] =
    typeof formatter == 'function'
      ? (txt) => /** @type {function} */ (formatter)(txt)
      : formatter;
  return map;
}, /**@type {Record<keyof typeof colors, (input?: *) => string>} */ ({}));
```

With this, all these invocations were free of type errors (no red squiggly lines):

image


In addition, there were zero errors returned by `amp check-build-system` due to `colors.js`.

WDYT?

File: build-system/tasks/release/index.js [hunk 2]
```diff
@@ -22,7 +22,7 @@ const fs = require('fs-extra');
 const klaw = require('klaw');
 const path = require('path');
 const tar = require('tar');
-const {cyan, green} = require('kleur/colors');
+const {cyan, green} = require('../common/colors');;
```
Comment (rsimha) [inline build-system/tasks/release/index.js:25]:
```suggestion
const {cyan, green} = require('../../common/colors');
```

**Review by rsimha — CHANGES_REQUESTED** [review]
Super glad you managed to figure this out! I've posted a blocking comment below because I think we should significantly simplify the wrapper. Lemme know what you think.

**Review by rsimha — COMMENTED** [review]
Can we also change line 28 in `babel.config.js`? (It's part of the compilation toolchain, but babel does not allow us to put it in `build-system/`.)

https://github.com/ampproject/amphtml/blob/df5b6a174c8270fcad5b4698fd53aae53f205e7f/babel.config.js#L28-L29

File: build-system/tasks/release/index.js (hunk 2 above)
Comment (rileyajones) [reply build-system/tasks/release/index.js:25]: Caught this myself before noticing your comment

File: babel.config.js [hunk 3]
```diff
@@ -25,7 +25,7 @@
 
 'use strict';
 
-const {cyan, yellow} = require('kleur/colors');
+const {cyan, yellow} = require('build-system/common/colors');
```
[the remark this answers is not in the document]
Comment (rileyajones) [reply babel.config.js:28]: In order to use this for babel we will have to rewrite using es5, do you think it's worth it?

**Review by rsimha — APPROVED** [review]
Yay! LGTM after the PR description is updated to reflect the latest changes.

**Pull request merged**